### PR TITLE
Danielle cantu

### PR DESCRIPTION
**Pull request by dlcantu**

### CODE REVIEW

File: code/danielle/Python/OOP.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    def display():
+        print(f"{Bank_Account}")
+
+Dani_account = Bank_Account("XXX-XX-3289", "Danielle C.", 345)
+Dani_account.deposit(20)
+Dani_account.withdrawl(40)
+Dani_account.bankFees()
+Dani_account.display
+
+print(Dani_account)
```
Comment (cdmichaelb) [inline code/danielle/Python/OOP.py]: [OOP lab](https://github.com/PdxCodeGuild/class_chihuahua/blob/main/1%20Python/labs/OOP.md) needs work:

Rectangle and Person Class missing.

File: code/danielle/Python/num_to_phrase_1.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    elif num in range(11, 20):
+        return teens_digit[num]
+    elif num in range(20, 100):
+        return (f"{tens_digit[(num//10)]} {ones_digit[(num%10)]}")
+        #return tens_digit[num] and ones_digit[num]
+        # tens_digit = x//10
+        # ones_digit = x%10
+
+num = (input("Please enter a number: "))
+print(num_to_phrase(int(num)))
```
Comment (cdmichaelb) [inline code/danielle/Python/num_to_phrase_1.py:17]: Good work on version 1 of Num to phrase.

File: code/danielle/Python/num_to_phrase_2.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        return (f"{tens_digit[(num//10)]} {ones_digit[(num%10)]}")
+    elif num in range(100, 999):
+        return (f"{hundreds_digit[(num//100)]} {tens_digit[(num//10)]} {ones_digit[(num%10)]}")
+
+num = (input("Please enter a number: "))
+print(num_to_phrase(int(num)))
+    # 
+
+    # elif num in range(100, 1000):
+    #     return (f"{hundreds_digit[(num//100)]} {tens_digit[(num//10)]} {ones_digit[(num%10)]}")
```
Comment (cdmichaelb) [inline code/danielle/Python/num_to_phrase_2.py]: Version 2 of [num to phrase](https://github.com/PdxCodeGuild/class_chihuahua/blob/main/1%20Python/labs/11A%20number_to_phrase.md#version-2) incomplete.

File: code/danielle/Python/unit_converter_4.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
+def unit_converter():
+    input_value = input("what is the distance?: ")
+    input_units = input("what are the input units? (ft, mi, km, yd, in): ")
+    output_units = input("What are the desired output units? (ft, mi, km, yd, in): ")
+    converted_units = {
```
Comment (cdmichaelb) [inline code/danielle/Python/unit_converter_4.py:5]: Wrong units are being converted. Should be ft, mi, m, and km. You can add extra if you want but should at least include these.

**Review by cdmichaelb — DISMISSED** [review]
Unit Converter v4, Num to Phrase v2, and OOP need work.

File: code/danielle/Python/API_Lab/API_lab_v2.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        print(f"The current weather is: {current_weather_data}")
+    elif Current_Weather == "N":
+        print(Five_Day)
+    if Five_Day == "Y":
+        print(f"The five day forecast is: {five_day_data}")
+    elif Five_Day == "N":
+        print(f"No data available")
+
+weather_API()
+
```
Comment (AlexDeMichieli) [inline code/danielle/Python/API_Lab/API_lab_v2.py:25]: A few issues:

1) It does not print any data
2) It looks like the data you're about to print is the entire JSON object. I recommend to be intentional with what you want to display, so you might want to extract from it.
3) The logic fails. The program asks if I want to get both five day or current weather

**Review by AlexDeMichieli — CHANGES_REQUESTED** [review]
Requests changes before merge

File: code/danielle/Python/num_to_phrase_2.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+         4: "four hundred", 5: "five hundred", 6: "six hundred", 
+         7: "seven hundred", 8: "eight hundred", 9: "nine hundred" }
+    if num in range(0, 11):
+        return ones_digit[num]
+    elif num in range(11, 20):
+        return teens_digit[num]
+    elif num in range(20, 100):
+        return (f"{tens_digit[(num//10)]} {ones_digit[(num%10)]}")
+    elif num in range(100, 999):
+        return (f"{hundreds_digit[(num//100)]} {tens_digit[(num//10)]} {ones_digit[(num%10)]}")
```
Comment (ttajohnson) [inline code/danielle/Python/num_to_phrase_2.py]: V2 still needs fix for 100+. Hint: Line 16, tens_digit math.

Comment (dlcantu) [reply code/danielle/Python/num_to_phrase_2.py]: I've been trying a few different things but I keep getting key errors. I don't really understand why the math works in line 14 but not 16. When I remove the tens_digit math from line 16, the hundreds_digit and ones_digit math works fine, but adding anything I try with the tens_digit gives me the key error in line 16.

**Review by AlexDeMichieli — APPROVED** [review]

**Pull request merged**